pyuvdata/uvdata/tests/test_mir_meta_data.py

```diff
@@ -954,11 +1192,53 @@ def test_mir_make_key_mask(mir_in_data, mir_bl_data, mir_sp_data):
 
     assert np.array_equal(mir_sp_data["sphid"], [6, 7, 8, 9, 10, 16, 17, 18, 19, 20])
 
-    # Finally, set the single ingtegration as bad, and make sure it cascades down
+    # Finally, set the single integration as bad, and make sure it cascades down
     mir_in_data._mask[:] = False
     mir_bl_data._make_key_mask(mir_in_data, reverse=True)
     mir_sp_data._make_key_mask(mir_bl_data, reverse=True)
 
     assert not any(mir_in_data._mask)
     assert not any(mir_bl_data._mask)
     assert not any(mir_sp_data._mask)
+
+
+def test_bad_codes(mir_codes_data, tmp_path):
+    filepath = os.path.join(tmp_path, "bad_codes")
+    bad_codes_copy = mir_codes_data.copy()
+
+    # Simulate a bad data entry, based on real life examples...
+    bad_codes_copy._data = bad_codes_copy._data.astype(mir_codes_data._binary_dtype)
+    bad_codes_copy._data[0][0] = b"filever\x00\xca"
+    bad_codes_copy.write(filepath)
+
+    new_codes = MirCodesData(filepath)
+    assert mir_codes_data == new_codes
+
+
+@pytest.mark.parametrize(
+    "has_file,kwargs,err_type,err_msg",
+    [
+        [None, {}, ValueError, "filepath must be of type str."],
+        [False, {}, FileNotFoundError, "No file found"],
+        [True, {"invert_check": True}, FileExistsError, "File already exists"],
+    ],
+)
+def test_gen_filepath_errs(tmp_path, has_file, kwargs, err_type, err_msg):
+    mir_meta_obj = MirMetaData(filetype="test")
+    if has_file:
+        with open(os.path.join(tmp_path, "test"), "a"):
+            pass
+
+    if has_file is not None:
+        # Make sure nothing happens if the check is disabled.
+        mir_meta_obj._gen_filepath(filepath=str(tmp_path), check=False, **kwargs)
+
+    print(tmp_path)
```

Comment:
Drop these prints?

Reply:
Removed in b36ea1b702e7a8f6a5a2c20b4fd257bf24c94afd.